source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
 
    sudo swupd update
 
-Install Kubernetes and CRI runtimes
-***********************************
+Learn about the benefits of having an up-to-date system for cloud
+orchestration on the :ref:`swupd-guide` page.
+
+
+Install Kubernetes
+******************
 
 Kubernetes, a set of supported :abbr:`CRI (Container Runtime Interface)`
-runtimes, and networking plugins, are included in the `cloud-native-basic`_
-bundle.
+runtimes and :abbr:`CNI (Container Network Interface)` are included in the
+`cloud-native-basic`_ bundle.
 
-To install this framework, enter the following command:
+#. Install the cloud-native-basic bundle to get these components:
 
 .. code-block:: bash
 
    sudo swupd bundle-add cloud-native-basic
 
-.. note::
 
-   For more on networking plugins, see `Install pod network add-on`_.
+Configure host system
+=====================
 
-Configure Kubernetes
-********************
+This tutorial uses the basic default Kubernetes configuration for to get
+started. You can customize your Kubernetes configuration according to your
+specific deployment and security needs.
 
-This tutorial uses the basic default Kubernetes configuration for simplicity.
-You must define your Kubernetes configuration according to your specific
-deployment and your security needs.
+The Kubernetes administration tool, :command:`kubeadm`, performs some
+"`preflight checks`_" when initializing and starting a cluster. The steps
+below are necessary to ensure those `preflight checks`_ pass successfully.
 
-#. Enable IP forwarding to avoid kubeadm `preflight check`_ errors:
 
-   Create (or edit if it exists) the file :file:`/etc/sysctl.d/60-k8s.conf`
-   and include the following line:
+#. Enable IP forwarding:
 
-   .. code-block:: bash
+   - Create the file :file:`/etc/sysctl.d/60-k8s.conf` to set the
+     :command:`net.ipv4.ip_forward` parameter
+
+     .. code-block:: bash
+
+        sudo mkdir -p /etc/sysctl.d/
+
+        sudo tee /etc/sysctl.d/99-kubernetes-cri.conf > /dev/null <<EOF
+        net.bridge.bridge-nf-call-iptables  = 1
+        net.ipv4.ip_forward                 = 1
+        net.bridge.bridge-nf-call-ip6tables = 1
+        EOF
+
+   - Apply the change:
+
+     .. code-block:: bash
+
+        sudo sysctl --system
 
-      net.ipv4.ip_forward = 1
 
-   Apply the change:
+#. Disable swap:
 
    .. code-block:: bash
 
-      sudo systemctl restart systemd-sysctl
+      sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
+      sudo swapoff -a
 
-#. Enable the kubelet service:
+   .. warning::
+
+      Kubernetes is designed to work without swap. Performance degradation of other workloads can occur
+      with swap disabled on systems with constrained memory resources.
+
+#. Add the the system's hostname to the :file:`/etc/hosts` file. Kubernetes
+   will read this file to locate the master host.
+
+    .. code-block:: bash
+
+       echo "127.0.0.1 localhost `hostname`" | sudo tee --append /etc/hosts
```

Comment:
This command if run twice adds the entry twice.

Reply:
True. We can make it replace the whole file, but there are many documents in the tree like this that kind of assume if you've done it once, you don't blindly repeat it, for sake of simplicity.